### ISSUE TITLE
Deno `task` regression in 1.21.1

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno#14459]
Hi. I just discover a bug in `deno task` command with additional arguments. I have the following tasks:

```json
{
  "tasks": {
    "build": "deno run -A https://deno.land/x/lume@v1.7.4/ci.ts",
    "serve": "deno task build -- -s"
  }
}
```

`deno task serve` runs the same script as `deno task build` but adding the `-s` argument (to start a server). This task works fine in Deno `1.21.0` but in `1.21.1` I get the following error:

```
$ deno task serve
Warning deno task is unstable and may drastically change in the future
Task serve deno task build -- -s
Warning deno task is unstable and may drastically change in the future
Task build deno run -A https://deno.land/x/lume@v1.7.4/ci.ts "--" "-s"
Check https://deno.land/x/lume@v1.7.4/ci.ts
error: Found argument '-s' which wasn't expected, or isn't valid in this context

        If you tried to supply `-s` as a value rather than a flag, use `-- -s`

USAGE:
    deno run [OPTIONS] <SCRIPT_ARG>...

For more information try --help
```

**Comment by dsherret** [comment]
The main issue here was caused by #14419 (deno task should have been forwarding `--`... it's unstable so we're still making these changes). The task needs to be changed to `"serve": "deno task build -s"`. That said, this is broken in 1.21.1 and fixed in #14434 which will be released on Thursday in 1.21.2 (I can't think of a workaround in 1.21.1).

Sorry about the hassle here! I know Lume auto-generates this for users so this will be a bit of a headache for a little bit.

I'm going to close this one as fixed in #14434.

**Comment by oscarotero** [comment]
Ok, understood.
I wanted to release a new version of Lume today but will wait for Deno 1.21.2 to update the task generation to the new syntax.
Thanks.
